packages/instantsearch-react/src/utils.js

```diff
@@ -12,3 +12,20 @@ export function isSpecialClick(event) {
 export function capitalize(key) {
   return key.length === 0 ? '' : `${key[0].toUpperCase()}${key.slice(1)}`;
 }
+
+export function assertFacetDefined(searchParameters, searchResults, facet) {
+  const wasRequested =
+    searchParameters.isConjunctiveFacet(facet) ||
+    searchParameters.isDisjunctiveFacet(facet);
```

Comment:
I'm not checking that hierarchical facets are defined yet. I don't think it would work quite the same way since the name of a hierarchical facet does not map to a facet.